src/m365/spo/commands/homesite/homesite-set.spec.ts

```diff
@@ -86,44 +91,189 @@ describe(commands.HOMESITE_SET, () => {
-    const requestBody = { sphSiteUrl: siteUrl, configuration: { vivaConnectionsDefaultStart: true } };
+  it('sets the specified site as the Home Site with targetedLicenseType to frontLineWorkers', async () => {
     const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
-      if (opts.url === `https://contoso-admin.sharepoint.com/_api/SPO.Tenant/SetSPHSiteWithConfiguration`) {
-        return vivaConnectionDefaultResponse;
+      if (opts.url === `${spoAdminUrl}/_api/SPO.Tenant/UpdateTargetedSite`) {
+        return defaultResponse;
       }
       return 'Invalid request';
     });
 
     await command.action(logger, {
       options: {
         siteUrl: siteUrl,
-        vivaConnectionsDefaultStart: true
+        targetedLicenseType: "frontLineWorkers"
+      }
+    } as any);
+
+    assert.deepStrictEqual(postRequestStub.lastCall.args[0].data, {
+      siteUrl: siteUrl,
+      configurationParam: { IsTargetedLicenseTypePresent: true, TargetedLicenseType: 1 }
+    });
+  });
+
+  it('sets the specified site as the Home Site with targetedLicenseType to informationWorkers', async () => {
+    const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${spoAdminUrl}/_api/SPO.Tenant/UpdateTargetedSite`) {
+        return defaultResponse;
+      }
+      return 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        siteUrl: siteUrl,
+        targetedLicenseType: "informationWorkers"
+      }
+    } as any);
+
+    assert.deepStrictEqual(postRequestStub.lastCall.args[0].data, {
+      siteUrl: siteUrl,
+      configurationParam: { IsTargetedLicenseTypePresent: true, TargetedLicenseType: 2 }
+    });
+  });
+
+  it('covers transformAudienceNamesToIds with multiple audience names', async () => {
+    // Mock entraGroup.getGroupIdByDisplayName to return different IDs for different names
+    const entraGroupStub = sinon.stub(entraGroup, 'getGroupIdByDisplayName');
+    entraGroupStub.withArgs('Marketing Team').resolves('00000000-0000-0000-0000-000000000001');
+    entraGroupStub.withArgs('Sales Team').resolves('00000000-0000-0000-0000-000000000002');
+
+    const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${spoAdminUrl}/_api/SPO.Tenant/UpdateTargetedSite`) {
+        return defaultResponse;
+      }
+      return 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        siteUrl: siteUrl,
+        audienceNames: 'Marketing Team, Sales Team'
       }
     } as any);
-    assert.deepStrictEqual(postRequestStub.lastCall.args[0].data, requestBody);
+
+    assert.deepStrictEqual(postRequestStub.lastCall.args[0].data, {
+      siteUrl: siteUrl,
+      configurationParam: { IsAudiencesPresent: true, Audiences: ['00000000-0000-0000-0000-000000000001', '00000000-0000-0000-0000-000000000002'] }
+    });
+
+    entraGroupStub.restore();
```

Comment:
Either we add this in the `afterEach` method, or we enclose this in a `finally`. Currently, this line will not be hit when this test fails.